server/apps/publish/content/common.py

```diff
@@ -157,6 +157,7 @@ def update(self, id, updates, original):
         try:
             user = get_user()
             last_updated = updates.get(config.LAST_UPDATED, utcnow())
+            auto_publish = updates.pop('auto_publish', False)
```

Comment:
it could also do `auto_publish = config.VERSION in updates` so it would make sure there is at least one version stored for any item published, but not sure it's any better..